### PR TITLE
Update webpage/theme selector 

### PR DESCRIPTION
**Pull request by vgoh1**
# Description 
The page/theme selector had some wack logic like not closing after clicking the cross or letting you add pages when there weren't anymore to add. Also did some touchups and refactoring of the Dashboard code.

# Type of change 
- [x] Bug fix
- [ ] New feature
- [x] Refactoring
- [ ] Documentation 

# Problem
See description

# Solution description
Fixed the wack logic mentioned above.
I refactored the webpage modals into different files: WebpageDelete and WebpageSelector.
Updated the image in TemplateSelector with the default deployed template, and it's not a modal anymore because I decided to use tabs. If anyone doesn't like tabs and want to go back to the card look, let me know... The initial idea was to have the deploy button in a third tab and also display the pages and theme selected but after talking to some people they prefer having the deploy button with the webpages selector. Also let me know what you guys think of that idea.
The TemplateSelector css probably needs to be fixed up but I'll leave that in Spring's capable hands since they'll be dealing with it.
Also refactored ConditionalWrapper into a component since it's being used more than once, to allow conditional tooltips.

# Tests 
See changes
image
image

# Relevant document/ link (if any) 
N/A

# Risk
Medium, dashboard is integral to the application

### CODE REVIEW

File: client/src/components/dashboard/deploy/DeployPage.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ import { Button, Spinner } from 'react-bootstrap';
 const DeployPage = ({ teamId }) => {
   const dispatch = useDispatch();
   const loading = useSelector((state) => state.deploy.loading);
+  // TODO: refactor this into useState
```
Comment (vgoh1) [inline client/src/components/dashboard/deploy/DeployPage.js:11]: The refactoring in this file will happen in another ticket as this PR is already too big

File: client/src/components/dashboard/Dashboard.js [hunk 2]
```diff
@@ -82,105 +31,38 @@ const Dashboard = () => {
...
-              currentWebPages={currentWebPages} 
-              directToAnotherPage={directToAnotherPage} 
-              showDeleteModal={showDeleteModal}
+        <Tabs
+          defaultActiveKey="home"
+          onSelect={(k) => setCurrentTab(k)}
+          transition={false}
+          className="mb-3"
+        >
+          <Tab eventKey="home" title="Webpages">
```
Comment (pmyip3) [inline client/src/components/dashboard/Dashboard.js:43]: can the eventKey named ''webpages' to make it consistent?

**Review by pmyip3 — APPROVED** [review]

File: client/src/components/dashboard/Dashboard.js (hunk 2 above)
Comment (vgoh1) [reply client/src/components/dashboard/Dashboard.js:43]: At least one of the tabs has to have an eventKey set to 'home' which is the default tab it'll start on when the page loads, if you don't have one set to 'home' there won't be a default tab selected and they'll have to click. See when eventKey is set to webpages:
image

File: client/src/components/shared/ConditionalWrapper.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+const ConditionalWrapper = ({ condition, wrapper, children }) =>
```
Comment (aksh0001) [inline client/src/components/shared/ConditionalWrapper.js:1]: File headers.

File: client/src/components/dashboard/webpage/WebpageSelector.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            className="float-right"
+            disabled={displayButton}
+            onClick={handleSubmit}
+          >
+            Confirm
+          </Button>
+        </Modal.Body>
+      </Modal>
+      
+
```
Comment (aksh0001) [inline client/src/components/dashboard/webpage/WebpageSelector.js:71]: blank lines

**Comment by aksh0001** [comment]
Please fix conflicts

**Review by rpra0006 — APPROVED** [review]

**Comment by vgoh1** [comment]
@aksh0001 fixed conflicts with main

**Review by jfan0010 — APPROVED** [review]

**Pull request merged**